katcp/core.py

```diff
@@ -687,20 +700,64 @@ def __init__(mcs, name, bases, dct):
         for name in dir(mcs):
             if not callable(getattr(mcs, name)):
                 continue
+            handler = getattr(mcs, name)
             if name.startswith("request_"):
                 request_name = convert_method_name("request_", name)
-                mcs._request_handlers[request_name] = getattr(mcs, name)
-                assert(mcs._request_handlers[request_name].__doc__ is not None)
+                if mcs.check_protocol(handler):
```

Comment:
yes, but then you need an additional if statement to check if any of the three cases were triggered, etc. I agonised about it for _minutes_ ;)